test/tools/WebListener/Controllers/CertController.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using mvc.Models;
+
+namespace mvc.Controllers
+{
+    public class CertController : Controller
+    {
+        public JsonResult Index()
+        {
+            // X509Certificate2 objects do not serialize as JSON. Create a HashTable instead
+            Hashtable output = new Hashtable 
+            {
+                {"Status", "FAILED"}
+            };
+            if (null != HttpContext.Connection.ClientCertificate)
+            {   
+                output = new Hashtable 
+                {
+                    {"Status"      , "OK"},
+                    {"Thumbprint"  , HttpContext.Connection.ClientCertificate.Thumbprint},
+                    {"Subject"     , HttpContext.Connection.ClientCertificate.Subject},
+                    {"SubjectName" , HttpContext.Connection.ClientCertificate.SubjectName.Name},
+                    {"Issuer"      , HttpContext.Connection.ClientCertificate.Issuer},
+                    {"IssuerName"  , HttpContext.Connection.ClientCertificate.IssuerName.Name},
+                    {"NotAfter"    , HttpContext.Connection.ClientCertificate.NotAfter},
+                    {"NotBefore"   , HttpContext.Connection.ClientCertificate.NotBefore}
+                };
```

Comment:
I still see ';'.

Reply:
The Semicolon has to be there. When I tried to remove it I get compile errors saying it is missing.

Reply:
Sorry. Closed.